kernel/src/hil/screen.rs

```diff
@@ -219,74 +241,63 @@ pub trait Screen<'a> {
         height: usize,
     ) -> Result<(), ErrorCode>;
 
-    /// Sends a write command to write data in the selected video memory frame.
-    /// When finished, the driver will call the `write_complete()` callback.
+    /// Write data from `buffer` to the selected write frame.
     ///
-    /// Return values:
-    /// - `Ok(())`: Write is valid and will be sent to the screen.
-    /// - `INVAL`: Write is invalid or length is wrong.
-    /// - `BUSY`: Another write is in progress.
-    fn write(&self, buffer: &'static mut [u8], len: usize) -> Result<(), ErrorCode>;
-
-    /// Sends a write command to write data in the selected video memory frame
-    /// without resetting the video memory frame position. It "continues" the
-    /// write from the previous position.
-    /// This allows using buffers that are smaller than the video mameory frame.
     /// When finished, the driver will call the `write_complete()` callback.
     ///
+    /// This function can be called multiple times if the write frame is larger
+    /// than the size of the available buffer. The write frame is only reset
+    /// when `set_write_frame()` is called.
+    ///
     /// Return values:
     /// - `Ok(())`: Write is valid and will be sent to the screen.
-    /// - `INVAL`: Write is invalid or length is wrong.
+    /// - `SIZE`: The buffer is too long for the selected write frame.
     /// - `BUSY`: Another write is in progress.
-    fn write_continue(&self, buffer: &'static mut [u8], len: usize) -> Result<(), ErrorCode>;
```

Comment:
I think we still need this. Some devices, like the st77xx screens need to know if this is a new write, in which case they start writing from the beginning of the memory, or if this is a continued write, and continue writing to the memory.

We had a longer discussion on this https://github.com/tock/tock/pull/3011#discussion_r878662467.

I might be missing something though.

Reply:
`write` now has a `continue_write` parameter, which I believe subsumes this method?